rust/src/lib/ip.rs

```diff
@@ -200,6 +200,40 @@ impl InterfaceIpv4 {
         Self::default()
     }
 
+    /// Set explicit default values if not desired. Used when do not want to
+    /// merge ip settings from current status
+    pub fn set_explicit_defaults(&mut self) {
+        // Set enabled as true when DHCP or addresses defined
+        if !self.enabled_defined
+            && (self.dhcp == Some(true)
+                || self
+                    .addresses
+                    .as_ref()
+                    .map(|a| !a.is_empty())
+                    .unwrap_or_default())
+        {
+            self.enabled = true
+        }
+
+        if self.dhcp.is_none() {
+            self.dhcp = Some(false)
+        }
```

Comment:
Does it make sense to enable dhcp if `enabled=false`?

Reply:
Removed. New approach been used.